roles/installer/tasks/upgrade_postgres.yml

```diff
@@ -0,0 +1,138 @@
+---
+
+# If postgres version is not 12, and not an external postgres instance (when managed_database is yes),
+# then run this playbook with include_tasks from database_configuration.yml
+# If _postgres_image == 'postgres'. Use pg_upgrade.  Otherwise, use POSTGRESQL_UPGRADE=copy support
+
+
+# Upstream (postgres image)
+
+- name: Scale down Deployment for migration
+  include_tasks: scale_down_deployment.yml
+
+- name: Delete existing postgres configuration secret
+  k8s:
+    api_version: v1
+    kind: Secret
+    name: "{{ ansible_operator_meta.name }}-postgres-configuration"
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    state: absent
+    wait: yes
+
+- name: Create Database configuration with new -postgres-13 hostname
+  k8s:
+    apply: true
+    definition: "{{ lookup('template', 'postgres_upgrade_secret.yaml.j2') }}"
+  no_log: true
+
+- name: Set new database var to be used when configuring app credentials (resources_configuration.yml)
+  set_fact:
+    awx_postgres_host: "{{ ansible_operator_meta.name }}-postgres-13"
+  no_log: true
+
+- name: Create Database if no database is specified
+  k8s:
+    apply: true
+    definition: "{{ lookup('template', 'postgres.yaml.j2') }}"
+    wait: true
+  register: create_statefulset_result
+
+- name: Get new postgres pod information
+  k8s_info:
+    kind: Pod
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    name: "{{ ansible_operator_meta.name }}-postgres-13-0"  # using name to keep compatibility
+    field_selectors:
+      - status.phase=Running
+  register: postgres_pod
+  until:
+    - "postgres_pod['resources'] | length"
+    - "postgres_pod['resources'][0]['status']['phase'] == 'Running'"
+  delay: 5
+  retries: 60
+
+- name: Set the resource pod name as a variable.
+  set_fact:
+    postgres_pod_name: "{{ postgres_pod['resources'][0]['metadata']['name'] }}"
+
+- name: Set full resolvable host name for postgres pod
+  set_fact:
+    resolvable_db_host: "{{ ansible_operator_meta.name }}-postgres.{{ ansible_operator_meta.namespace }}.svc.cluster.local"  # yamllint disable-line rule:line-length
+  no_log: true
+
+- name: Set pg_dump command
+  set_fact:
+    pgdump: >-
+      pg_dump
+      -h {{ resolvable_db_host }}
+      -U {{ awx_postgres_user }}
+      -d {{ awx_postgres_database }}
+      -p {{ awx_postgres_port }}
+      -F custom
+  no_log: true
+
+- name: Set pg_restore command
+  set_fact:
+    pg_restore: >-
+      pg_restore
+      -U {{ awx_postgres_user }}
+      -d {{ awx_postgres_database }}
+  no_log: true
+
+- name: Stream backup from pg_dump to the new postgresql container
+  k8s_exec:
+    namespace: "{{ ansible_operator_meta.namespace }}"
+    pod: "{{ postgres_pod_name }}"
+    command: |
+      bash -c """
+      set -e -o pipefail
+      PGPASSWORD={{ awx_postgres_pass }} {{ pgdump }} | PGPASSWORD={{ awx_postgres_pass }} {{ pg_restore }}
```

Comment:
I'm not sure if the k8s_exec module supports passing env variables but it would be great to define `PGPASSWORD` as an env var rather than on the cli (if possible)

Reply:
@dsavineau unfortunately it does not support it.  If I add the following to that task, it sets it inside the operator container which is executing the role's tasks, not within the postgresql container which is where the env var is needed.  

```
  environment:
    PGPASSWORD: {{ awx_postgres_pass }}
```